tests/main/econnreset/task.yaml

```diff
@@ -14,13 +14,13 @@ execute: |
     rm -f test-snapd-huge_*
     su -c "/usr/bin/env SNAPD_DEBUG=1 snap download --edge test-snapd-huge 2>snap-download.log" test &
 
-    echo "Wait until the download started and downloaded more than 1 MB"
+    echo "Wait until the download started"
     for _ in $(seq 40); do
         partial=$(find . -name 'test-snapd-huge_*.snap.partial' | head -1)
-        if [ -n "$partial" ] && [ "$(stat -c%s "$partial")" -gt "$(( 1024 * 1024 ))" ]; then
+        if [ -n "$partial" ] && [ "$(stat -c%s "$partial")" -gt "$(( 1 ))" ]; then
```

Comment:
This has increased the raciness of the test significantly instead of reducing it. It has also changed the nature of the test in that what's now being tested is the assertion retrying instead of the download retrying. Which is also good, but now what this test was meant to cover.

Another hint that this isn't having the supposed effect is that you're *reducing* the size it checks for. If the download was indeed very fast, why would we reduce the size? We have more than half a gig in the file being downloaded.

Do you have a run of this test with logs to show what the actual flow is?

What happens if the sleep below is cut down to .1, without changing anything else in the test?

Reply:
This loop waits for the download to start before applying iptables rule. The waiting is realized by observing the size of the file as it's being downloaded in the background. I think there is no benefit of waiting for >1Mb. Waiting for just a single byte is actually increasing the window for the actual test and lessens the raciness, IMHO. Reducing the sleep down increases the chances of capturing the right moment when the download started as well.

No disagreement about changing the nature of the test (download retry vs download or assertion retry), I'll drop this change.